src/lib/is_array.js

```diff
@@ -18,10 +18,20 @@ var dv = (typeof DataView === 'undefined') ?
     function() {} :
     DataView;
 
-exports.isTypedArray = function(a) {
+function isTypedArray(a) {
     return ab.isView(a) && !(a instanceof dv);
-};
+}
+
+function isArrayOrTypedArray(a) {
+    return Array.isArray(a) || isTypedArray(a);
+}
+
+function is1D(a) {
+    return !isArrayOrTypedArray(a[0]);
```

Comment:
Maybe `isArray1D` would be clearer as this assumes that you're passing an array (or at least something you can index in). 

We might want to add a comment above about that assumption too, in case someone in the future wants to turn this into:

```js
return Array.isArray(a) && !isArrayOrTypedArray(a[0]);
```

Reply:
https://github.com/plotly/plotly.js/pull/2577/commits/2a41f9ed9184014a543004ae6a5042334ccf8fec -> `isArray1D`